### DISCUSSION TITLE
Support Managing CI/CD Variables Dynamically

### DISCUSSION THREAD

**Discussion by ashrafgt** [community/community#9249] [Actions]
It seems that Github already supports creating secrets from the UI, but could this be extended to manage non-secret variables?

**Comment by ethomson** [comment]
Good question - at the moment you can add secrets via the UI (as you mentioned) or you can add non-secret [environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables) to a workflow.  What sorts of limitations are you running into with workflow variables that would be solved with adding variables to the UI?

**Reply by ashrafgt** [reply]
I think the limitation is that the codebase has to be modified in order to update the variables. Perhaps it's because I'm used to Gitlab CI variables that it feels like one.

In case you haven't used them much before, Gitlab CI variables, which are defined dynamically via the UI or the API, can be:
1. Visible or Hidden (Github currently supports the latter, dynamically, that is)
2. Variable or File (Github currently supports the former, statically, that is)

**Reply by mdvorak** [reply]
I second this, there is a reason why Kubernetes have Secret and ConfigMap objects. There are plenty of configuration items, should would be nice to have defined like secret, but not being masked.
I've seen many times things like hostname being defined as secret, simply because there is no other way.

**Comment by GlatBane** [comment]
it looks like this is now being planned https://github.com/github/roadmap/issues/575

**Comment by codebydant** [comment]
Related to https://github.com/community/community/discussions/42055
